src/store/singing.ts

```diff
@@ -145,6 +268,40 @@ const singingVoiceCache = new Map<SingingVoiceSourceHash, SingingVoice>();
 
 const initialTrackId = TrackId(crypto.randomUUID());
 
+export const singingStorePlugin: WatchStoreStatePlugin = (store) => {
+  // tracksの変更とtrackChannelStripsを同期する。
+  // NOTE: ChannelStripをVuex内に入れると変更された扱いで警告が出るため、
+  // Vuex内に入れない。
```

Comment:
ここのコメントってどういうことでしたっけ 👀 

watch内でmutation経由せずVuexのstateを変えると警告が出てしまう、的な？
だとするとmutationで変更すれば良いのかってなってリファクタリングしちゃおうとする人が出てくるかも

こういうロジック･････？（自信なし）

```suggestion
  // NOTE: ChannelStripはCommandに未対応なのでVuex.stateに入れられない。
```

Reply:
> watch内でmutation経由せずVuexのstateを変えると警告が出てしまう、的な？

何も触ってないのに変更された扱いになっていた記憶があります。（忘れた）

Reply:
> （忘れた）

メモの意味が･･･！！

流石に何も触ってなければ大丈夫だとは思うので、setするときになんか起こるって感じですかねぇ。

まあせっかくなので再検証してVueやImmerのことをちょっとわかるようになるか、面倒であれば「NOTE: ChannelStripはVuexのstateに入れられないのでwatchで同期する」とか書けば良さそう？

Reply:
どうやらVuexに入れる事自体は問題なさそうです。
問題はimmerが検知しないのでUndo/Redoが動かないことなので、そっちをコメントに書くようにします。
[immer-please-work.diff.txt](https://github.com/user-attachments/files/16312024/immer-please-work.diff.txt)